### PR TITLE
add functionality to skip running codestyle where desired

### LINKED ISSUE #32: Implement an exclusion path list
I'm working on a project that has a series of autogenerated source files that, sadly, do not conform to our desired guidelines. In order to prevent excessive changes to these autogenerated files, I would like to exclude them from being subjected to the codestyle tests.

I have a working patch and can make a pull request if this seems like a desirable feature.

### PR DESCRIPTION
**Pull request by mattwwarren**
Fixes #32

### CODE REVIEW

**Comment by mattwwarren** [comment]
No worries. I also discovered a global exclusion in pytest applies to coverage, so I have a workaround for now.

Re: tests - I think it's complaining about a blank line between imports but that empty line does not exist in my diff?

**Comment by henry0312** [comment]
Maybe, you didn't push the latest commit.

image

**Comment by mattwwarren** [comment]
https://github.com/henry0312/pytest-codestyle/pull/33/commits/9713c1db7721c3ae34edbbcdf5ad2c6601bfcf36 is the last commit I have in my clone and shows no whitespace between lines 1 and 2

Test output reads:
```
=================================== FAILURES ===================================
_________________________________ isort-check __________________________________
ERROR: /home/travis/build/henry0312/pytest-codestyle/pytest_codestyle.py Imports are incorrectly sorted.
 import fnmatch
+
 import py.io
 import pycodestyle
 import pytest
===================== 1 failed, 17 passed in 0.66 seconds ======================
```

Did I misread it? Should there be a blank line?

File: pytest_codestyle.py [hunk 1]
```diff
@@ -38,6 +41,9 @@ def setup(self):
         mtime = self.fspath.mtime()
         if old_mtime == mtime:
             pytest.skip('previously passed pycodestyle checks')
+        for path in self.config.getini('codestyle_exclude'):
```
Comment (henry0312) [inline pytest_codestyle.py]: I think these should be in `pytest_collect_file`.

File: pytest_codestyle.py [hunk 2]
```diff
@@ -18,11 +20,14 @@ def pytest_addoption(parser):
...
     parser.addini('codestyle_show_source', type="bool", default=True,
                   help='show source code for each error (default: True)')
+    parser.addini('codestyle_exclude', type="pathlist", default=[],
+                  help='source files to exclude from codestyle')
 
 
 def pytest_collect_file(parent, path):
     config = parent.config
-    if config.option.codestyle and path.ext == '.py':
+    print(path)
```
Comment (henry0312) [inline pytest_codestyle.py]: please delete this `print`

Comment (mattwwarren) [reply pytest_codestyle.py]: Ah! I knew I wrote that and totally forgot. I'm the worst.

**Pull request merged**